src/core/Akka.Remote/Transport/Helios/HeliosTransport.cs

```diff
@@ -475,7 +476,32 @@ public static EndPoint AddressToSocketAddress(Address address)
         public static Address MapSocketToAddress(IPEndPoint socketAddr, string schemeIdentifier, string systemName, string hostName = null)
         {
             if (socketAddr == null) return null;
-            return new Address(schemeIdentifier, systemName, hostName ?? socketAddr.Address.ToString(), socketAddr.Port);
+            return new Address(schemeIdentifier, systemName, SafeMapHostName(hostName) ?? SafeMapIPv6(socketAddr.Address), socketAddr.Port);
+        }
+
+        /// <summary>
+        /// Check to see if a given hostname is IPV6, IPV4, or DNS and apply the appropriate formatting
+        /// </summary>
+        /// <param name="hostName">The hostname parsed from the file</param>
+        /// <returns>An RFC-compliant formatting of the hostname</returns>
+        public static string SafeMapHostName(string hostName)
+        {
+            IPAddress addr;
+            if (!string.IsNullOrEmpty(hostName) && IPAddress.TryParse(hostName, out addr))
+            {
+                if (addr.AddressFamily == AddressFamily.InterNetworkV6)
+                    return "[" + addr + "]";
```

Comment:
Need to wrap any IPV6 address in `[` and `]`, per the RFC. Otherwise `Uri` can't parse it from a fully qualified `ActorPath`